base_cron_oneshot/tests/test_cron.py

```diff
@@ -0,0 +1,84 @@
+
+from odoo import api
+from odoo.modules.registry import Registry
+from odoo.tests import common
+from contextlib import contextmanager
+
+
+class OneshotTestCase(common.TransactionCase):
+
+    @property
+    def cron_model(self):
+        return self.env['ir.cron']
+
+    def test_defaults(self):
+        cron = self.cron_model.create({
+            'oneshot': True,
+            'name': 'Foo',
+            'model_id': self.env['ir.model']._get('ir.cron').id,
+            'state': 'code',
+            'code': 'model.some_method()',
+            'interval_number': 1,
+            'interval_type': 'days',
+            'numbercall': 5,  # won't have any effect
+        })
+        self.assertRegexpMatches(cron.name, 'Oneshot#\d+ Foo')
+        self.assertEqual(cron.numbercall, 1)
+
+    def test_schedule_oneshot_check(self):
+        with self.assertRaises(AssertionError) as err:
+            self.cron_model.schedule_oneshot('res.partner')
+        self.assertEqual(str(err.exception), 'Provide a method or some code!')
+
+    def test_schedule_oneshot_method(self):
+        cron = self.cron_model.schedule_oneshot('res.partner', method='read')
+        self.assertRegexpMatches(cron.name, 'Oneshot#\d+')
+        self.assertEqual(cron.numbercall, 1)
+        self.assertEqual(cron.code, 'model.read()')
+        self.assertEqual(
+            cron.model_id, self.env['ir.model']._get('res.partner'))
+
+    def test_schedule_oneshot_code(self):
+        cron = self.cron_model.schedule_oneshot(
+            'res.partner', code='env["res.partner"].search([])')
+        self.assertRegexpMatches(cron.name, 'Oneshot#\d+')
+        self.assertEqual(cron.numbercall, 1)
+        self.assertEqual(cron.code, 'env["res.partner"].search([])')
+        self.assertEqual(
+            cron.model_id, self.env['ir.model']._get('res.partner'))
+
+
+class OneshotProcessTestCase(common.TransactionCase):
+
+    def setUp(self):
+        super().setUp()
+        deleted = []
+
+        @api.multi
+        def unlink(self):
+            deleted.extend(self.ids)
+            return unlink.origin(self)
+
+        self.env['ir.cron']._patch_method('unlink', unlink)
+        self.addCleanup(self.env['ir.cron']._revert_method, 'unlink')
+        self.deleted = deleted
+
+    @contextmanager
+    def _cron_fake_cursors(self):
+        job_cr = Registry(common.get_db_name()).cursor()
+        cron_cr = Registry(common.get_db_name()).cursor()
+        yield job_cr, cron_cr
+        job_cr.rollback()
+        job_cr.close()
+        cron_cr.rollback()
+        cron_cr.close()
+
+    def test_schedule_oneshot_unlink(self):
+        cron = self.env['ir.cron'].schedule_oneshot(
+            'res.partner', method='read')
+        cron_data = cron.read(load='_classic_write')[0]
+        with self._cron_fake_cursors() as (job_cr, cron_cr):
+            self.env['ir.cron']._process_job(job_cr, cron_data, cron_cr)
+        self.assertIn(cron.id, self.deleted)
```

Comment:
Wouldn't it be simpler to just search on `ir.cron` and make sure `cron` is not there anymore ?